cmd/ztest.c

```diff
@@ -7966,6 +7966,9 @@ exec_child(char *cmd, char *libpath, boolean_t ignorekill, int *statusp)
 				VERIFY3S(-1, !=,
 				    asprintf(&newlp, "%s:%s", libpath, curlp));
 				VERIFY0(setenv("LD_LIBRARY_PATH", newlp, 1));
+				if (newlp != NULL) {
+					free(newlp);
+				}
```

Comment:
We don't need the conditional check here.  We already verify above that the `asprintf()` succeeded and then user that value in `setenv()`, but even if we didn't `free()` accepts NULL as a no-op input.

Reply:
Good point. I just pushed a revision that calls `free(newlp)` unconditionally.

> even if we didn't free() accepts NULL as a no-op input.

I am not sure if that is true across glibc, musl libc, uclibc, freebsd libc, etcetera, but with the VERIFY statement, we do not need to worry about this. :)

Reply:
> I am not sure if that is true across glibc, musl libc, uclibc, freebsd libc, etcetera, but with the VERIFY statement, we do not need to worry about this. :)

I used to fret about exactly the same thing, but in practice this behavior is quite universal these days.